Buildings/Controls/OBC/Utilities/PIDWithAutotuning/FirstOrderAMIGO.mo

```diff
@@ -0,0 +1,503 @@
+          0}}, color={0,0,127}));
+
+annotation (Documentation(info="<html>
+<p>
+This block implements a rule-based PID tuning method.
+</p>
+<p>
+The PID tuning method approximates the control process with a
+first-order time delayed (FOTD) model.
+It then determines the gain and delay of this FOTD model based on the responses of
+the control process to an asymmetric relay feedback.
+After that, taking the gain and delay of this FOTD mode as inputs, this PID tuning
+method calculates the PID gains with an Approximate M-constrained Integral Gain Optimization (AMIGO) Tuner.
+</p>
+<p>
+This block is implemented using
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithInputGains\">
+Buildings.Controls.OBC.Utilities.PIDWithInputGains</a>
+and inherits most of its configuration. However, through the parameter
+<code>controllerType</code>, the controller can only be configured as PI or
+PID controller.
+</p>
+
+<h4>Autotuning Process</h4>
+<p>
+To use this block, place it in an control loop as any other PID controller.
+Before the PID tuning process starts, this block is equivalent to
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithInputGains\">
+Buildings.Controls.OBC.Utilities.PIDWithInputGains</a>.
+This block starts the PID tuning process when a request for performing autotuning occurs, i.e.,
+when the value of the boolean input signal <code>triTun</code> changes from
+<code>false</code> to <code>true</code>.
+During the autotuning process, the output of the block changes into that of a relay controller
+(see <a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Controller\">
+Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Controller</a>).
+The PID tuning process ends automatically
+(see details in
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.BaseClasses.TuningMonitor\">
+Buildings.Controls.OBC.Utilities.PIDWithAutotuning.BaseClasses.Relay.TunMonitor</a>),
+at which point this block turns back to a PID controller but with tuned PID parameters.
+</p>
+
+<p>
+<b>Note:</b> If an autotuning is ongoing, i.e., <code>inTunPro.y = true</code>,
+a request for performing autotuning will be ignored.
+</p>
+<h4>Guidance for settings the parameters</h4>
+<p>
+The performance of the autotuning is affected by the parameters, including the
+typical range of control error, <code>r</code>, 
+the reference output for the tuning process, <code>yRef</code>, the lower value for
+the relay output, <code>yLow</code>, and the deadband, <code>deaBan</code>.
+<br>
+The following procedure can be used for determining the values of those parameters. 
+</p>
+<ol>
+<li>
+The <code>r</code> should be adjusted so that the output of the relay controller,
+<code>rel.yDif</code>, is within the range from 0 to 1.
+</li>
+<li>
+The <code>yRef</code> can be determined by dividing the set point by the sum of the
```

Comment:
How do use know the set point and the minimum and maximum values of the measurement? Both the set point and measurement are inputs.
Do you mean the typical values?